### PR TITLE
#1550 allow passing appconfig to vt during init

### LINKED ISSUE #1550: Allow users to pass AppConfig to configuration VT during inititialization
**What Needs to be Done?**

Simpler version of https://github.com/DARMA-tasking/vt/issues/444

### PR DESCRIPTION
**Pull request by jstrzebonski**
Fixes #1550 

---
TODO:
- [x] print options overwritten by CLI arguments
- [x] clean up ArgConfig struct - if possible move member functions to the anonymous namespace in *.cc file
- [x] fix unit tests failures (gcc-10, ubuntu, openmpi, no LB)
- [x] Add documentation
- [x] Add documentation on appConfig being overwritten by CLI arguments

### CODE REVIEW

**Comment by jstrzebonski** [comment]
Printing options predefined in source code, that are overwritten by CLI arguments:

image

image

File: tests/unit/runtime/test_initialization.cc [hunk 1]
```diff
@@ -49,38 +49,111 @@
...
+  vt::initialize(custom_argc, custom_argv, no_workers, true, &comm, &appConfig);
 
-TEST_F(TestInitialization, test_initialize_with_args) {
   EXPECT_EQ(theConfig()->prog_name, "vt_program");
+  EXPECT_EQ(theConfig()->vt_color, false);
+  EXPECT_EQ(theConfig()->vt_epoch_graph_on_hang, false);
+  EXPECT_EQ(theConfig()->vt_lb_name, "RotateLB");
+  EXPECT_EQ(theConfig()->vt_lb_stats, true);
   EXPECT_EQ(theConfig()->vt_no_terminate, true);
+  // CLI args should overwritten hardcoded appConfig
```
Comment (cz4rs) [inline tests/unit/runtime/test_initialization.cc:155]: `should overwrite` / `should have overwritten`

**Comment by cz4rs** [comment]
@jstrzebonski This functionality should be mentioned in documentation and the precedence should be described explicitly (`test_initialization.cc` helps too :+1:).

File: tests/unit/runtime/test_initialization.cc (hunk 1 above)
Comment (jstrzebonski) [reply tests/unit/runtime/test_initialization.cc]: Done.

**Review by cz4rs — CHANGES_REQUESTED** [review]
Please add a short description into docs (basically a more user-friendly way to tell users what you wrote in unit tests - `  // CLI args should overwrite hardcoded appConfig`).

File: src/vt/configs/arguments/args.cc [hunk 2]
```diff
@@ -56,21 +56,141 @@ namespace vt { namespace arguments {
...
+    args_to_parse.push_back(*it);
+  }
+
+  // Allow a input config file
+  app.set_config(
+    "--vt_input_config",
+    "", // no default file name
+    "Read in an ini config file for VT",
+    false // not required
+  );
```
Comment (cz4rs) [inline src/vt/configs/arguments/args.cc:130]: This was not added in this PR, but I wonder whether this works / is used somewhere?

Comment (jstrzebonski) [reply src/vt/configs/arguments/args.cc:130]: Documented and covered by unit tests.

**Comment by jstrzebonski** [comment]
CLI11 parses configuration file before command-line arguments. I added appropriate documentation and two new unit tests covering this behavior.

**Review by cz4rs — APPROVED** [review]
👍

**Review by lifflander — APPROVED** [review]
Looks good to me!

**Comment by nlslatt** [comment]
Note that the automatic rebase feature doesn't retain signatures. I'll fix it.

**Comment by jstrzebonski** [comment]
> Note that the automatic rebase feature doesn't retain signatures. I'll fix it.

No problem, I've just force-pushed.

**Pull request merged**